core/src/main/scala/org/apache/spark/executor/ProcfsMetricsGetter.scala

```diff
@@ -37,7 +37,9 @@ private[spark] case class ProcfsMetrics(
     pythonVmemTotal: Long,
     pythonRSSTotal: Long,
     otherVmemTotal: Long,
-    otherRSSTotal: Long)
+    otherRSSTotal: Long,
+    timeStamp: Long)
```

Comment:
I think it will be better if we keep the timestamp metric outside of ProcfsMetrics, as this is not a metric and I could not find any benefit of keeping inside ProcfsMetrics

Reply:
I considered timestamp as a property of the set of metrics, so probably better to keep it like this.

Reply:
I think responded to that. What is the point of separating it when it is used as a property of the set of metrics?

Reply:
I think the point that both I and Imran are trying to make here is that timestamp is not a metric and thus it should not be kept inside ProcfsMetrics. The purpose of timestamp is to evict older cached metrics and thus it should be a part of ProcfsMetricsGetter along with `cachedAllMetric`.

Reply:
In my view when I introduced ProcfsMetrics, I considered it as a class. The object of this class have properties which are metrics and now a timestamp. So I think it is better to have that property defined along with others. This make the code cleaner as well.  So sorry if I can't understand why it shouldn't be here.